### PR TITLE
PUBDEV-5284 - Enable zip64 for shadowJar

### PR DESCRIPTION
**Pull request by Pscheidl**

### CODE REVIEW

File: h2o-assemblies/main/build.gradle [hunk 1]
```diff
@@ -49,7 +49,8 @@ for (comp in optionalComponents) {
     
 }
 
-shadowJar {
+shadowJar { 
+  zip64 true
```
Comment (michalkurka) [inline h2o-assemblies/main/build.gradle]: How come this was not needed before? We have tests that verify importing ORC files

Comment (mmalohlava) [reply h2o-assemblies/main/build.gradle]: it is used already in hadoop assembly. we probably hit limit, when pavel was debugging ORC and include ORC into default main assembly.

**Comment by Pscheidl** [comment]
Let me summarize just hadoop-* dependencies:

```text
== :h2o-assemblies:main: Using optional component: jython-cfunc, version 3.17.0
:h2o-persist-hdfs:dependencyInsight
org.apache.hadoop:hadoop-annotations:2.6.0-cdh5.4.0
+--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
|    \--- compile
\--- org.apache.hadoop:hadoop-common:2.6.0-cdh5.4.0
     +--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
     \--- org.apache.hadoop:hadoop-aws:2.6.0-cdh5.4.0
          +--- compile
          \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)

org.apache.hadoop:hadoop-auth:2.6.0-cdh5.4.0
\--- org.apache.hadoop:hadoop-common:2.6.0-cdh5.4.0
     +--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
     |    \--- compile
     \--- org.apache.hadoop:hadoop-aws:2.6.0-cdh5.4.0
          +--- compile
          \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)

org.apache.hadoop:hadoop-aws:2.6.0-cdh5.4.0
+--- compile
\--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
     \--- compile

org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
\--- compile

org.apache.hadoop:hadoop-common:2.6.0-cdh5.4.0
+--- org.apache.hadoop:hadoop-aws:2.6.0-cdh5.4.0
|    +--- compile
|    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
|         \--- compile
\--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)

org.apache.hadoop:hadoop-hdfs:2.6.0-cdh5.4.0
\--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
     \--- compile

org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0
\--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
     \--- compile

org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0
+--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0
|    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
|         \--- compile
+--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0
|    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
\--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0
     +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0 (*)
     \--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0 (*)

org.apache.hadoop:hadoop-mapreduce-client-core:2.6.0-cdh5.4.0
+--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
|    \--- compile
\--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0
     +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0
     |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
     +--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0
     |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
     \--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0
          +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0 (*)
          \--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0 (*)

org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0
\--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
     \--- compile

org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0
+--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0
|    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
|         \--- compile
\--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0
     \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)

org.apache.hadoop:hadoop-yarn-api:2.6.0-cdh5.4.0
+--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
|    \--- compile
+--- org.apache.hadoop:hadoop-yarn-client:2.6.0-cdh5.4.0
|    \--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0
|         +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0
|         |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
|         +--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0
|         |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
|         \--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0
|              +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0 (*)
|              \--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0 (*)
+--- org.apache.hadoop:hadoop-yarn-common:2.6.0-cdh5.4.0
|    +--- org.apache.hadoop:hadoop-mapreduce-client-core:2.6.0-cdh5.4.0
|    |    +--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
|    |    \--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0 (*)
|    +--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0 (*)
|    +--- org.apache.hadoop:hadoop-yarn-client:2.6.0-cdh5.4.0 (*)
|    +--- org.apache.hadoop:hadoop-yarn-server-common:2.6.0-cdh5.4.0
|    |    +--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0 (*)
|    |    +--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0 (*)
|    |    \--- org.apache.hadoop:hadoop-yarn-server-nodemanager:2.6.0-cdh5.4.0
|    |         \--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0 (*)
|    \--- org.apache.hadoop:hadoop-yarn-server-nodemanager:2.6.0-cdh5.4.0 (*)
+--- org.apache.hadoop:hadoop-yarn-server-common:2.6.0-cdh5.4.0 (*)
\--- org.apache.hadoop:hadoop-yarn-server-nodemanager:2.6.0-cdh5.4.0 (*)

org.apache.hadoop:hadoop-yarn-client:2.6.0-cdh5.4.0
\--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0
     +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0
     |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
     |         \--- compile
     +--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0
     |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
     \--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0
          +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0 (*)
          \--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0 (*)

org.apache.hadoop:hadoop-yarn-common:2.6.0-cdh5.4.0
+--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0
|    +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0
|    |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
|    |         \--- compile
|    +--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0
|    |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
|    \--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0
|         +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0 (*)
|         \--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0 (*)
+--- org.apache.hadoop:hadoop-mapreduce-client-core:2.6.0-cdh5.4.0
|    +--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
|    \--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0 (*)
+--- org.apache.hadoop:hadoop-yarn-client:2.6.0-cdh5.4.0
|    \--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0 (*)
+--- org.apache.hadoop:hadoop-yarn-server-common:2.6.0-cdh5.4.0
|    +--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0 (*)
|    +--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0 (*)
|    \--- org.apache.hadoop:hadoop-yarn-server-nodemanager:2.6.0-cdh5.4.0
|         \--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0 (*)
\--- org.apache.hadoop:hadoop-yarn-server-nodemanager:2.6.0-cdh5.4.0 (*)

org.apache.hadoop:hadoop-yarn-server-common:2.6.0-cdh5.4.0
+--- org.apache.hadoop:hadoop-mapreduce-client-common:2.6.0-cdh5.4.0
|    +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0
|    |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
|    |         \--- compile
|    +--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0
|    |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)
|    \--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0
|         +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0 (*)
|         \--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0 (*)
+--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0 (*)
\--- org.apache.hadoop:hadoop-yarn-server-nodemanager:2.6.0-cdh5.4.0
     \--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0 (*)

org.apache.hadoop:hadoop-yarn-server-nodemanager:2.6.0-cdh5.4.0
\--- org.apache.hadoop:hadoop-mapreduce-client-shuffle:2.6.0-cdh5.4.0
     +--- org.apache.hadoop:hadoop-mapreduce-client-app:2.6.0-cdh5.4.0
     |    \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
     |         \--- compile
     \--- org.apache.hadoop:hadoop-mapreduce-client-jobclient:2.6.0-cdh5.4.0
          \--- org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0 (*)

(*) - dependencies omitted (listed previously)
```

**Comment by Pscheidl** [comment]
However, the includeOrc boolean only affects this dependency:

```groovy
    def hadoopVersion = project.hasProperty("doIncludeOrc") && project.doIncludeOrc == "true" ?
                        orcDefaultHadoopClientVersion : defaultHadoopClientVersion
    compile("org.apache.hadoop:hadoop-client:$hadoopVersion") {
        // Pull all dependencies to allow run directly from IDE or command line
        transitive = true
    }
```

The dependency of hadoop-client with includeOrc on is following:

```text
org.apache.hadoop:hadoop-client:2.6.0-cdh5.4.0
\--- compile
```

File: h2o-assemblies/main/build.gradle (hunk 1 above)
Comment (michalkurka) [reply h2o-assemblies/main/build.gradle]: Yep, I want to make sure that enabling the ORC doesn't bring unwanted dependencies.

What I mean is enabling it is fine but we should also look what dependencies get us over the limit.

**Comment by Pscheidl** [comment]
We have 24 MB of new dependencies, which is strange !

**Pull request merged**